Add Docs for HCP Connectivity Tester (#2996)
hashicorp/web-unified-docs: content/hcp-docs/content/docs/vault/connectivity-tester.mdx

## Supported tests

| Test | `type` enum value | CLI equivalent | What it checks |
|---|---|---|---|
| **DNS lookup** | `CONNECTIVITY_COMMAND_TYPE_DIG` | `dig +noall +answer <hostname> @<nameserver>` | Resolves a hostname using your DNS server and returns the resolved IP addresses |
| **TCP reachability** | `CONNECTIVITY_COMMAND_TYPE_NC` | `nc -z -v <hostname> <port>` | Attempts a TCP connection to a hostname and port and reports open, closed, or unreachable |
| **TLS / curl** | `CONNECTIVITY_COMMAND_TYPE_CURL` | `curl https://<hostname>:<port>` | Performs a TLS handshake to confirm a TLS-capable service is reachable |
| **Ping (ICMP)** | `CONNECTIVITY_COMMAND_TYPE_PING` | `ping -c 5 <hostname>` | Sends ICMP echo requests to verify basic host reachability |
| **Traceroute** | `CONNECTIVITY_COMMAND_TYPE_TRACEROUTE` | `traceroute <hostname>` | Maps the network path between the Vault cluster and the target host |

## Limitations

> - The connectivity tester is available for HCP Vault Dedicated clusters only.

Reviewer: There are still a handful of passive voice mentions in the doc - write in passive voice - "is <something>" is a good indicator its passive voice.  Some generally easy tricks to remove it, make it about the user "You can use...", move the noun up (which is in the suggestion), or flip the sentance - Only HVD clusters support the.."
```suggestion
- The connectivity tester supports HCP Vault Dedicated clusters only.
```

> - The connectivity tester is scoped to the primary HVN. After a backup region is promoted, connectivity to features such as BYO-DNS or PrivateLink is not retained in the promoted region.

Reviewer: ```suggestion
- The connectivity tester supports the clusters primary HVN. After a backup region is promoted, connectivity to features such as BYO-DNS or PrivateLink is not retained in the promoted region.
```

# START AUTO GENERATED METADATA, DO NOT EDIT
created_at: 2026-08-04T13:57:53-05:00
last_modified: 2026-08-04T19:28:07+05:30
# END AUTO GENERATED METADATA
---

# HCP Vault Connectivity Tester

The HCP Connectivity Tester is a self-service diagnostic tool that verifies
network reachability from your HCP Vault Dedicated cluster to internal
resources, including databases, identity providers, LDAP servers, and DNS
resolvers. It lets you confirm that DNS resolution, TCP reachability, and
network connectivity are working correctly across your HashiCorp Virtual Network
(HVN). Use it before configuring Vault integrations such as database secrets
engines, LDAP authentication, or MFA providers.

Without this tool, network failures only surface after you deploy a Vault
configuration, producing vague error logs that require engineering escalation.
The connectivity tester gives you clear, structured diagnostics similar to
common CLI diagnostic commands — equivalent to running `dig`, `nc`, `curl`,
`ping`, or `traceroute` — without needing to interpret raw logs or configure
Vault first.

## Prerequisites

- Access to the [HCP Portal](https://portal.cloud.hashicorp.com/sign-in).

> - AWS HCP Vault Dedicated Cluster created in the configured AWS HVN.

Reviewer 2: ```suggestion
- AWS HCP Vault Dedicated cluster created in the configured AWS HVN.
```

   $ curl --location "https://api.cloud.hashicorp.com/vault/2020-11-25/organizations/$HCP_ORG_ID/projects/$HCP_PROJ_ID/clusters/$HCP_CLUSTER_ID/test-connectivity" \
    --request POST \
    --header 'Content-Type: application/json' \
    --header "Authorization: Bearer $HCP_API_TOKEN" \
    --data '{
      "connection_info": [
        {
          "type": "CONNECTIVITY_COMMAND_TYPE_DIG",
          "hostname": "database.internal.corp"
        }
      ]
    }' | jq
   ```

</Tab>
</Tabs>

## API reference

The HCP Vault Dedicate connectivity tester API provides an endpoint for
initiaing tests. There are different tests available based on various network
scenarios.

### Versioning

The HCP Vault Dedicated API version is `2023-05-01`.

### Prerequisites

- [Retrieve an access token](/hcp/docs/hcp/api#authenticate-to-hcp) for authentication.

### Test connectivity

`POST` `/vault/2020-11-25/organizations/{organization_id}/projects/{project_id}/clusters/{cluster_id}/test-connectivity`

The request body accepts a `connection_info` array. Each element is a flat object with a `type` enum and the parameters for that test. You can submit one or more tests in a single request.

```text

> POST /vault/2020-11-25/organizations/{organization_id}/projects/{project_id}/clusters/{cluster_id}/test-connectivity

Reviewer 2: Since this is a cURL command, use `shell-session` instead of `text` ([style guide](https://github.com/hashicorp/web-unified-docs/blob/main/docs/style-guide/codeblocks-and-consoles/organization.md#examples)).

```suggestion
```shell-session
$ POST /vault/2020-11-25/organizations/{organization_id}/projects/{project_id}/clusters/{cluster_id}/test-connectivity
```

| `connection_results[].error` | object \| null | `null` on success. Contains `code` and `message` on failure. |
| `connection_results[].error.code` | string | Machine-readable error code. See [Error codes](#error-codes) below. |
| `connection_results[].error.message` | string | Human-readable error description. |
| `connection_results[].duration_ms` | string | Elapsed time for the test in milliseconds. |

<Note>

The blocklist is enforced at both input validation and post-DNS-resolution stages. This prevents DNS rebinding attacks where a hostname initially passes validation but resolves to a blocked internal IP at runtime.

</Note>

## Audit logging

Every connectivity test invocation is recorded in the HCP audit log with the following fields:

- Caller identity (`organization_id`, `project_id`, user token subject)
- Invocation timestamp
- Cluster ID
- Test result (`CONNECTION_STATUS_SUCCESS` or `CONNECTION_STATUS_FAILURE`)

## Troubleshooting

The results of the connectivity test help you determine what troubleshooting
steps to consider. 

### `HOSTNAME_UNRESOLVED` returned

Verify that:
- The target hostname exists in your private DNS zone.

> - If using a custom resolver_ip (`DIG` only), confirm the resolver_ip is correct and reachable from the HVN.

Reviewer 2: ```suggestion
- If using a custom `resolver_ip` (`DIG` only), confirm the `resolver_ip` is correct and reachable from the HVN.
```
